### DISCUSSION TITLE
ZodType Class does not infer branded opaque preperties

### DISCUSSION THREAD

**Discussion by Nurai1** [colinhacks/zod#2000] [Q&A] [answered]
I have function `validateApiData` and it is a generic function that infer type from zod parser.
<img width="462" alt="image" src="https://user-images.githubusercontent.com/45640899/217201158-2a8490fd-389d-463d-a2da-8678dc3ee93b.png">

Below is an example how I can use it with the parser with the branded ID property.
<img width="545" alt="image" src="https://user-images.githubusercontent.com/45640899/217201623-28ce4f3b-be3f-40b6-959c-0d189cd0ab0b.png">

But id string has inferred prop type "string", not branded id.
<img width="323" alt="image" src="https://user-images.githubusercontent.com/45640899/217201773-a1a33953-2f1c-45e3-8d0a-8f7d425177ad.png">

I want this type to be like this
<img width="317" alt="image" src="https://user-images.githubusercontent.com/45640899/217202700-4dae4303-e9b0-45cc-9712-27511bd7c122.png">

Maybe you need to add it in next releases?

**Comment by JacobWeisenburger** [comment, marked as answer]
The code you are showing seems to have some other library that I'm not familiar with. Please use code blocks instead of pictures and give an example with only zod code that shows the issue you are having. Otherwise, it's really hard to understand what you are asking for. Thanks
